extensions/positron-python/python_files/posit/positron/convert.py

```diff
@@ -125,83 +148,98 @@ def _convert_column_filters(
+        """Generate code for sorting.
 
         Returns
         -------
         tuple[List[StrictStr], List[StrictStr]]
-            Tuple containing preprocessing and method chain code strings.
+            Tuple of (method_chain_setup, method_chain_parts).
         """
-        raise NotImplementedError("Subclasses must implement _convert_sorts method")
+        if not self.params.sort_keys:
+            return [], []
 
+        handler = self.sort_handler_class(
+            self.params.sort_keys, self.table, was_series=self.was_series
+        )
+        method_chain_setup, method_chain_parts = handler.convert_sorts()
 
-class PandasConverter(CodeConverter):
-    def __init__(
-        self, table, table_name: str, params: ConvertToCodeParams, *, was_series: bool = False
-    ):
-        self.was_series = was_series
-        super().__init__(table, table_name, params)
+        return method_chain_setup, method_chain_parts
 
-    def convert(self) -> List[StrictStr]:
-        if self.params.code_syntax_name.code_syntax_name != "pandas":
-            raise NotImplementedError(
-                f"Code conversion for {self.params.code_syntax_name} is not implemented."
-            )
-        return super().convert()
+    def _format_filter(self, comparisons: List[str]) -> tuple[List[StrictStr], List[StrictStr]]:
+        """Format multiple filter comparisons into method chain parts.
 
-    def _convert_row_filters(self) -> tuple[List[StrictStr], List[StrictStr]]:
-        """Take filters and convert them to code strings.
+        Parameters
+        ----------
+        comparisons : List[str]
+            List of comparison strings to format.
 
         Returns
         -------
         tuple[List[StrictStr], List[StrictStr]]
-            Tuple of (method_chain_setup, method_chain_parts).
+            Tuple of setup and method chain parts.
         """
-        method_chain_setup = []
-        method_chain_parts = []
-        filters = self.params.row_filters
+        raise NotImplementedError("Subclasses must implement _format_filter method")
 
-        if not filters:
-            return method_chain_setup, method_chain_parts
 
-        comparisons = []
-        # process each filter key to some comparison string
-        # that can be used in the method chain
-        for filter_key in filters:
-            handler = PandasFilterHandler(filter_key, self.table_name, was_series=self.was_series)
-            comparison = handler.convert_filters()
-            if comparison:
-                comparisons.append(comparison)
+class PandasConverter(CodeConverter):
+    def __init__(
+        self, table, table_name: str, params: ConvertToCodeParams, *, was_series: bool = False
+    ):
+        super().__init__(
+            table,
+            table_name,
+            params,
+            was_series=was_series,
+            filter_handler_class=PandasFilterHandler,
+            sort_handler_class=PandasSortHandler,
```

Comment:
Could just be class variables (or properties):

```python
class PandasConverter(CodeConverter):
    filter_handler_class = PandasFilterHandler
    sort_handler_class = PandasSortHandler
```

Reply:
updated!